apps/web/components/ServiceInformation/ServiceInformation.tsx

```diff
@@ -1,117 +1,281 @@
+        return {
+          label: x.version.split('-').pop(),
+          value: x.xroadIdentifier,
+        }
+      })
+    : [
+        {
+          label: n('noVersion'),
+          value: {
+            instance: '',
+            memberClass: '',
+            memberCode: '',
+            serviceCode: '',
+            subsystemCode: '',
+          },
+        },
+      ]
+
+  const [selectedVersionOption, setSelectedVersionOption] = useState<
+    SelectOption
+  >(versionOptions[0])
+
+  const [serviceDetail, setServiceDetail] = useState<ServiceDetail>(
+    service.environments[0].details[0],
+  )
+
+  const onSelectVersion = (versionOption: SelectOption) => {
+    const tempServiceDetail = service.environments
+      .find((e) => e.environment === selectedEnviromentOption.value)
+      .details.find((e) => e.xroadIdentifier === versionOption.value)
+
+    setServiceDetail(tempServiceDetail)
+    setSelectedVersionOption(versionOption)
+    onSelectChange(tempServiceDetail)
+  }
+
+  const onSelectEnviroment = (enviromentOption: SelectOption) => {
+    //TODO this function needs to be finished similar to onSelectVersion
+    setServiceDetail(
+      service.environments
+        .find((e) => e.environment === enviromentOption.value)
+        .details.find(selectedVersionOption.value),
+    )
+    setSelectedEnviromentOption(enviromentOption)
+  }
+
   return (
     <Box>
-      <Inline space={1}>
-        <Text variant="h1" as="h1">
-          {service.title}
+      <Box marginTop={1} marginBottom={3}>
+        <Box display="flex" alignItems="flexStart">
+          <Text variant="h1" as="h1">
+            {service.title}
+          </Text>
+          {service.pricing.length > 0 && (
+            <Box marginLeft={1}>
+              <Tag>{n(`pricing${capitalize(service.pricing[0])}`)}</Tag>
+            </Box>
+          )}
+        </Box>
+        <Text variant="intro" paddingTop={2}>
+          {service.description}
         </Text>
-        {service.pricing.length > 0 && (
-          <Box>
-            <Tag variant="white" outlined>
-              {n(`pricing${capitalize(service.pricing[0])}`)}
-            </Tag>
+      </Box>
+
+      <Box paddingBottom={3}>
+        <Divider />
+      </Box>
+
+      <Box
+        display="flex"
+        flexDirection={['column', 'row']}
+        justifyContent={['flexStart', 'spaceBetween']}
+      >
+        <Box display="flex">
+          <Box className={styles.selectDesktop}>
+            <Select
+              backgroundColor="blue"
+              size="sm"
+              label={n('XroadIdentifierInstance')}
+              name="Instance"
+              disabled={enviromentOptions.length < 2}
+              isSearchable={false}
+              defaultValue={selectedEnviromentOption}
+              options={enviromentOptions}
+              onChange={onSelectEnviroment}
+            />
+          </Box>
+          <Box marginLeft={2} className={styles.selectDesktop}>
```

Comment:
yes I tried styling it with boxes and grids and the select component was very hard to style. So It will be like this for now. It will be "Þróun", "Prófun" and "Raun" so it will probably fit

Reply:
We must at least make it fit with 'Development' (not all caps) because when the page is translated that will probably be the translation for "Þróun"

Reply:
I have fixed this :)